src/index.spec.ts

```diff
@@ -63,4 +63,11 @@ describe("assign", () => {
       }
     });
   });
+
+  it("should throw when unsafe path encountered", () => {
+    const payload = JSON.parse(
```

Comment:
You don't really need `JSON.parse` to test this, it's just a plain object.

Reply:
changed accordingly! Please have a look...